pkg/buildpack/buildpack.go

```diff
@@ -326,6 +327,183 @@
 	return layerTar, nil
 }
 
+func ToNLayerTar(dest string, module BuildModule) ([]ModuleTar, error) {
+	modReader, err := module.Open()
+	if err != nil {
+		return nil, errors.Wrap(err, "opening blob")
+	}
+	defer modReader.Close()
+
+	tarCollection := newModuleTarCollection(dest)
+	tr := tar.NewReader(modReader)
+
+	var (
+		header     *tar.Header
+		forWindows bool
+	)
+
+	for {
+		header, err = tr.Next()
```

Comment:
Handled at line 353

Reply:
Perhaps we don't need to do it as part of this PR (because we use tar readers all over) but it would be cleaner to wrap `tar.NewReader` with our own reader that sanitizes the path as part of calling `tr.Next()`